### PR TITLE
Text container element

### PR DESCRIPTION
**Pull request by raphaelm**

### CODE REVIEW

File: libpretixprint/src/main/java/eu/pretix/libpretixprint/templating/Layout.java [hunk 1]
```diff
@@ -23,6 +24,7 @@
 
 import static com.lowagie.text.Utilities.millimetersToInches;
 import static com.lowagie.text.Utilities.millimetersToPoints;
+import static eu.pretix.libpretixprint.templating.TextUtilsKt.linesRequiredHardBreak;
```
Comment (wiffbi) [inline libpretixprint/src/main/java/eu/pretix/libpretixprint/templating/Layout.java]: I cannot say much about the actual code, but reading it I think this method should be named `linesRequireHardBreak` (without the d) or `linesHardBreakRequired`.

File: libpretixprint/src/main/java/eu/pretix/libpretixprint/templating/Layout.java [hunk 2]
```diff
@@ -177,27 +196,133 @@ private void drawTextarea(JSONObject data, String text, PdfContentByte cb) throw
...
+                   3000000,
+                    (float) para.fontSize,
+                    para.alignment
+            );
+            ct.go(true);
+            realHeight = 3000000 - ct.getYLine();
+            realWidth = ct.getFilledWidth();
+
+            var smallEnough = (realHeight <= height && realWidth <= width);
+            if (!data.optBoolean("splitlongwords", true) && linesRequiredHardBreak(text, para.font, width + widthextension)) {
```
Comment (wiffbi) [inline libpretixprint/src/main/java/eu/pretix/libpretixprint/templating/Layout.java]: See comment above, I think it reads better (and semantically seems more right) to call the method `linesRequireHardBreak`

File: libpretixprint/src/main/java/eu/pretix/libpretixprint/templating/TextUtils.kt [hunk 3]
```diff
@@ -0,0 +1,136 @@
+package eu.pretix.libpretixprint.templating
+
+import com.lowagie.text.Font
+import com.lowagie.text.Utilities
+import com.lowagie.text.pdf.BaseFont
+import com.lowagie.text.pdf.BidiLine
+import com.lowagie.text.pdf.PdfChunk
+import com.lowagie.text.pdf.PdfWriter
+
+fun linesRequiredHardBreak(content: String, font: Font, rectangularWidth: Float): Boolean {
```
Comment (wiffbi) [inline libpretixprint/src/main/java/eu/pretix/libpretixprint/templating/TextUtils.kt]: See comment above

**Review by wiffbi — COMMENTED** [review]
I am not into Java/Kotlin so cannot say much about the code, but when reading, I stumbled upon the name of `linesRequiredHardBreak`, which I would call `linesRequireHardBreak`. Other than that looks good to me, but again, not really any experience with Java/Kotlin

**Review by robbi5 — APPROVED** [review]

**Pull request merged**